pkg/kubelet/dockershim/network/kubenet/kubenet_linux.go

```diff
@@ -243,45 +288,67 @@ func (plugin *kubenetNetworkPlugin) Event(name string, details map[string]interf
 		return
 	}
 
-	klog.V(5).Infof("PodCIDR is set to %q", podCIDR)
-	_, cidr, err := net.ParseCIDR(podCIDR)
-	if err == nil {
-		setHairpin := plugin.hairpinMode == kubeletconfig.HairpinVeth
-		// Set bridge address to first address in IPNet
-		cidr.IP[len(cidr.IP)-1] += 1
+	klog.V(4).Infof("kubenet: PodCIDR is set to %q", podCIDR)
+	podCIDRs := strings.Split(podCIDR, ",")
 
-		json := fmt.Sprintf(NET_CONFIG_TEMPLATE, BridgeName, plugin.mtu, network.DefaultInterfaceName, setHairpin, podCIDR, cidr.IP.String())
-		klog.V(2).Infof("CNI network config set to %v", json)
-		plugin.netConfig, err = libcni.ConfFromBytes([]byte(json))
-		if err == nil {
-			klog.V(5).Infof("CNI network config:\n%s", json)
+	// reset to one cidr if dual stack is not enabled
+	if !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.IPv6DualStack) && len(podCIDRs) > 1 {
+		klog.V(2).Infof("This node has multiple pod cidrs assigned and dual stack is not enabled. ignoring all except first cidr")
+		podCIDRs = podCIDRs[0:1]
+	}
 
-			// Ensure cbr0 has no conflicting addresses; CNI's 'bridge'
-			// plugin will bail out if the bridge has an unexpected one
-			plugin.clearBridgeAddressesExcept(cidr)
+	for idx, currentPodCIDR := range podCIDRs {
+		_, cidr, err := net.ParseCIDR(currentPodCIDR)
+		if nil != err {
+			klog.Warningf("Failed to generate CNI network config with cidr %s at indx:%v: %v", currentPodCIDR, idx, err)
+			return
 		}
-		plugin.podCidr = podCIDR
-		plugin.gateway = cidr.IP
+		// create list of ips and gateways
+		cidr.IP[len(cidr.IP)-1] += 1 // Set bridge address to first address in IPNet
+		plugin.podCIDRs = append(plugin.podCIDRs, cidr)
+		plugin.podGateways = append(plugin.podGateways, cidr.IP)
 	}
 
+	//setup hairpinMode
+	setHairpin := plugin.hairpinMode == kubeletconfig.HairpinVeth
+
+	json := fmt.Sprintf(NET_CONFIG_TEMPLATE, BridgeName, plugin.mtu, network.DefaultInterfaceName, setHairpin, plugin.getRangesConfig(), zeroCIDRv4, zeroCIDRv6)
+	klog.V(4).Infof("CNI network config set to %v", json)
+	plugin.netConfig, err = libcni.ConfFromBytes([]byte(json))
 	if err != nil {
-		klog.Warningf("Failed to generate CNI network config: %v", err)
+		klog.V(2).Infof("** failed to set up CNI with %v err:%v", json, err)
```

Comment:
Any reason to change this log message from Warningf (which I think is the right level) to an Infof()?)